npc/custom/stylist.txt

```diff
@@ -3,32 +3,68 @@
 //===== By: ==================================================
 //= Euphy
 //===== Current Version: =====================================
-//= 1.1
+//= 1.2
 //===== Description: =========================================
 //= Changes your hair style, hair color, and cloth color.
 //===== Additional Comments: =================================
 //= 1.1 Switched to 'getbattleflag', credits to Saithis. [Euphy]
+//= 1.2 Fix style start at min_style, add Job_Summoner [AnnieRuru]
 //============================================================
 
 prontera,170,180,1	script	Stylist#custom_stylist	2_M_DYEINGER,{
-
-	setarray .@styles[1],getbattleflag("max_cloth_color"),getbattleflag("max_hair_style"),getbattleflag("max_hair_color");
-	setarray .@Look[1],7,1,6;
-	set .@s, select(" ~ Cloth color", " ~ Hairstyle", " ~ Hair color");
-	set .@Revert, getlook(.@Look[.@s]);
-	set .@style,1;
-	while(1) {
-		setlook .@Look[.@s], .@style;
-		message strcharinfo(PC_NAME),"This is style #"+.@style+".";
-		set .@menu$, " ~ Next (^0055FF"+((.@style!=.@styles[.@s])?.@style+1:1)+"^000000): ~ Previous (^0055FF"+((.@style!=1)?.@style-1:.@styles[.@s])+"^000000): ~ Jump to...: ~ Revert to original (^0055FF"+.@Revert+"^000000)";
-		switch(select(.@menu$)) {
-			case 1: set .@style, ((.@style!=.@styles[.@s])?.@style+1:1); break;
-			case 2: set .@style, ((.@style!=1)?.@style-1:.@styles[.@s]); break;
-			case 3: message strcharinfo(PC_NAME),"Choose a style between 1 - "+.@styles[.@s]+".";
-				input .@style,0,.@styles[.@s];
-				if (!.@style) set .@style, rand(1,.@styles[.@s]);
-				break;
-			case 4: set .@style, .@Revert; setlook .@Look[.@s], .@Revert; break;
+	.@choose = select("Hair style", "Hair color", "Cloth color") - 1;
+	.@part = .look[.@choose];
+	if (BaseClass != Job_Summoner)
+		callsub(L_styles, .@part, .minstyle[.@part], .maxstyle[.@part]);
+	else
+		callsub(L_styles, .@part, .summoner_minstyle[.@part], .summoner_maxstyle[.@part]);
```

Comment:
missing `end;` ?
```suggestion
		callsub(L_styles, .@part, .summoner_minstyle[.@part], .summoner_maxstyle[.@part]);
	end;
```

Reply:
https://github.com/rathena/rathena/blob/master/npc/custom/battleground/bg_emp.txt#L79

you see, when I actually submit this script to Euphy,
I consider a little bit about changing my usual scripting style to suit rathena general behavior

```
OnRedDown:
	callsub L_EmpDown, "Red", .blue;
	end
OnBlueDown:
	callsub L_EmpDown, "Blue", .red;
	end;
L_EmpDown:
	mapannounce "bat_a01", strcharinfo(0) +" has destroyed "+ getarg(0) +" Team's Emperium.", bc_map;
	.winside = getarg(1);
	awake strnpcinfo(0);
	return;
```
this feels like OnRedDown and OnBlueDown has its own operation and nothing to do with each other
```
OnRedDown:  callsub L_EmpDown, "Red", .blue; end;
OnBlueDown: callsub L_EmpDown, "Blue", .red; end;
L_EmpDown:
	mapannounce "bat_a01", strcharinfo(0) +" has destroyed "+ getarg(0) +" Team's Emperium.", bc_map;
	.winside = getarg(1);
	awake strnpcinfo(0);
	return;
```
I am wondering what the `end;` is doing there

in the end, this 1 line method was accepted not only by Euphy, but also lots of rAthena developers
https://github.com/rathena/rathena/pull/3025

I'm quite sure if the `callsub` was meant never has a return; there's no need for an end; there